pipeline/test/services/service-cluster/testOpensearch.sh

```diff
@@ -0,0 +1,426 @@
+    no_error=true
+    debug_msg=""
+
+    default_policies_enabled=$(yq4 -e '.opensearch.ism.defaultPolicies' "${config['config_file_sc']}")
+    if [[ $default_policies_enabled == "true" ]]; then
+        default_policies=("kubernetes" "kubeaudit" "authlog" "other")
+        for policy in "${default_policies[@]}"; do
+            res=$(curl -w "%{http_code}" -o /dev/null -sk -u admin:"${adminPassword}" -X GET "https://opensearch.${opsDomain}/_plugins/_ism/policies/${policy}")
+            if [[ $res != "200" ]]; then
+                no_error=false
+                debug_msg+="[ERROR] Missing default policy : ${policy}\n"
+            fi
+        done
+    fi
+
+    # additional_policies=$(yq4 -e '.opensearch.ism.additionalPolicies | keys' "${config['config_file_sc']}")
+    readarray -t additional_policies < <(yq4 e -o=j -I=0 '.opensearch.ism.additionalPolicies | keys' "${config['config_file_sc']}")
+
+    additional_policies=("${additional_policies//,/ }")
+    additional_policies=("${additional_policies##[}")
+    additional_policies=("${additional_policies%]}")
+    # additional_policies=($additional_policies)
+    read -ra additional_policies <<<"${additional_policies[@]}"
+
+    for policy in "${additional_policies[@]}"; do
+        policy_name=$(echo "$policy" | awk -F'.' '{print $1}' | tr '"' ' ')
+        res=$(curl -w "%{http_code}" -o /dev/null -sk -u admin:"${adminPassword}" -X GET "https://opensearch.${opsDomain}/_plugins/_ism/policies/${policy_name}")
+        if [[ $res != "200" ]]; then
+            no_error=false
+            debug_msg+="[ERROR] Missing additional policy : ${policy_name}\n"
+        fi
+    done
+
+    indices_data=$(curl -sk -u admin:"${adminPassword}" -X GET "https://opensearch.${opsDomain}/_cat/indices" | awk '{print $3}' | tr '\n' ' ')
+    IFS=' ' read -ra indices <<<"$indices_data"
+
+    for index in "${indices[@]}"; do
+        index_data=$(curl -sk -u admin:"${adminPassword}" -X GET "https://opensearch.${opsDomain}/$index")
+        index_alias=$(echo "$index_data" | jq -r ".\"$index\".settings.index.plugins.index_state_management.rollover_alias")
+        is_write_index=$(echo "$index_data" | jq -r ".\"$index\".aliases.$index_alias.is_write_index")
+
+        if [[ "$is_write_index" == "true" ]]; then
+            rollover_age_days=$(yq4 -e '.opensearch.ism.rolloverAgeDays' "${config['config_file_sc']}")
+            epoch_now=$(date +%s)
+            index_creation_date=$(echo "$index_data" | jq -r ".\"$index\".settings.index.creation_date")
+
+            index_creation_date_seconds=$(echo "scale=3; $index_creation_date/1000" | bc)
+            rollover_diff=$(echo "scale=3; ($epoch_now-$index_creation_date_seconds)/86400" | bc | awk '{printf "%f", $0}')
+            if (($(echo "$rollover_diff > $rollover_age_days" | bc -l))); then
+                no_error=false
+                echo -e "[ERROR]The write index : $index is old and has not been rolled over as expected\n"
+            fi
```

Comment:
Is there a reason why you calculate divisions here?

I'd suggest sticking to integers and use the smallest unit if possible, because the `printf` and `bc` combo becomes locale dependant and doesn't work for me. :sweat_smile:

Reply:
I've changed it now so I can work with milliseconds directly